### DISCUSSION TITLE
Limits on expressiveness?

### DISCUSSION THREAD

**Discussion by glicerico** [Attempto/APE#41] [Q&A] [answered]
Are there limits to the expressiveness of ACE? Or, in other words, can ACE express as much coherent meaning as Natural English?

**Comment by tkuhn** [comment, marked as answer]
The semantics of ACE maps to first-order logic, so you cannot express anything with it that requires higher-order logic. For example, a full representation of "most" requires second-order logic, and therefore cannot be expressed in ACE. ACE covers a large part of first-order logic, but has limits on nesting and arity of relations, so there can be statements in first-order logic that cannot be expressed in ACE.

Moreover, ACE has also some small syntactic extensions that don't have a mapping to first order logic, most importantly for negation as failure ("not provably"), to make ACE usable for example in rule systems.

I hope that answers your question?

**Reply by glicerico** [reply]
Question answered, thanks!
